src/_interfaces/sampler.js

```diff
@@ -20,8 +20,22 @@
 // THE SOFTWARE.
 
 declare interface Sampler {
-    isSampled(operation: string): boolean;
+    name(): string;
+    
+    /**
+     * Decides if a new trace starting with given operation name
+     * should be sampled or not. If the method returns true, it
+     * must populate the tags dictionary with tags that identify
+     * the sampler, namely sampler.type and sampler.param.
+     *
+     * @param {string} operation - Operation name of the root span.
+     * @param {Object} tags - output dictionary to store sampler tags.
```

Comment:
I am trying to minimize the number of object allocations. Javascript does not support multi-value functions, so I would need to return something like `{sampled: bool, tags: map}` objects, which (a) need to be allocated, and (b) in most cases will say sampled=false where tags are not relevant.